### DISCUSSION TITLE
Crash on startup with Wayland fractional scaling

### DISCUSSION THREAD

**Discussion by StackDoubleFlow** [ppy/osu-framework#6216] [Q&A] [answered]
I'm running the following Sway revision:
- Sway: [2e95116](https://github.com/swaywm/sway/commit/2e95116)
- wlroots: [86e1b24d8](https://gitlab.freedesktop.org/wlroots/wlroots/-/commit/86e1b24d8)

The SampleGame crashes immediately on startup with `wp_viewport@33: error 2: source rectangle out of buffer bounds`. This happens on osu-lazer as well.

This issue only seems to happen with the following line in my Sway config: `output eDP-1 scale 1.5`. Here is my full sway config:
[config.txt](https://github.com/ppy/osu-framework/files/14602508/config.txt)

Here are the full logs with `WAYLAND_DEBUG=1`: 
[wayland-osu.log](https://github.com/ppy/osu-framework/files/14602521/wayland-osu.log)

I noticed the same issue happens on Factorio as well, so it may be an upstream SDL or Sway issue, not too sure.

**Comment by bdach** [comment]
Please report upstream.

**Reply by StackDoubleFlow** [reply, marked as answer]
I have reported upstream at https://github.com/libsdl-org/SDL/issues/9283
